packages/arb-validator/chainobserver/chainObserver.go

```diff
@@ -351,11 +353,26 @@ func (chain *ChainObserver) HandleNotification(ctx context.Context, event arbbri
 	case arbbridge.AssertedEvent:
 		chain.notifyAssert(ctx, ev)
 	case arbbridge.ConfirmedEvent:
-		return chain.confirmNode(ctx, ev)
+		if err := chain.confirmNode(ctx, ev); err != nil {
+			return err
+		}
+	}
+	chain.currentEventId = arbbridge.ChainInfo{
+		BlockId:  event.GetChainInfo().BlockId.Clone(),
+		LogIndex: event.GetChainInfo().LogIndex + 1,
 	}
 	return nil
 }
 
+func (chain *ChainObserver) NotifyNextEvent(blockId *common.BlockId) {
```

Comment:
Is this thread-safe?  If multiple threads are trying to do this at the same time, whichever one goes last will set the currentEventId.  But it's not clear that currentEventId makes sense as a concept if multiple threads are trying to update this.  

This might well be OK.  If so, suggest adding a comment saying why.

Reply:
All the public functions of ChainObserver are thread safe. There's a mutex that's locked over this whole function